common4j/src/main/com/microsoft/identity/common/java/BuildValues.java

```diff
@@ -0,0 +1,56 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.java;
+
+import javax.annotation.Nonnull;
+
+//Used as a wrapper for setting and accessing values through either the generated BuildConfig.java class,
+//or from parameters set via the NativeAuthPublicClientApplicationConfiguration.kt file.
+public class BuildValues {
```

Comment:
I remember mentioning that you should remove this class and either use BuildConfig directly or leverage the AadSliceInfo

Reply:
I'll let Saurabh run this conversation further, but AFAIK this was not the decision. We can't just rely on AadSliceInfo, as we also need a flag for useRealAuthority. And that flag needs to be controllable by a calling app (e.g. sample app), so BuildConfig won't be suited for that.
@SaurabhMSFT please let me know if I've missed anything. This is an important topic to align on.

Reply:
I have created a PBI to track this work item. We will be doing it in next quarter.

https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2707234

Reply:
@shahzaibj are you sure the field you mean is called AadSliceInfo? A ctrl-F for that brings up 0 results in the project.